model/common/tests/common/states/unit_tests/test_model_options.py

```diff
@@ -0,0 +1,35 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+from icon4py.model.common.model_backends import BACKENDS
+from icon4py.model.common import dimension as dims
+from icon4py.model.common.model_options import customize_backend
+from icon4py.model.common.type_alias import vpfloat
+from icon4py.model.common.utils import data_allocation as data_alloc
+from icon4py.model.common.grid.simple import simple_grid
+from icon4py.model.testing import test_utils
+import numpy as np
+
+
+def test_custom_backend():
+    backend = "run_gtfn_cpu_cached"
+    match_backend = [
+        backend_str.split("_")
+        for backend_str in BACKENDS.keys()
+        if set(backend_str.split("_")).issubset(set(backend.split("_")))
+    ][0]
+    backend_options = {
+        "device": match_backend[1],
+        "backend_kind": match_backend[0],
+    }
```

Comment:
for this test, why not just
```suggestion
    backend = "run_gtfn_cpu_cached"
    match_backend = [
        backend_str.split("_")
        for backend_str in BACKENDS.keys()
        if set(backend_str.split("_")).issubset(set(backend.split("_")))
    ][0]
    backend_options = {
        "device": "cpu", # or whatever it will be after the discussion
        "backend_kind": "gtfn", # or the corresponding maker if we go that way
    }
```

Reply:
it just extrapolates info from backends that we already have. I thought it might be useful to keep it. Does it bother you?

Reply:
I don't exactly understand what you want to test, so my answer depends on your answer to that question.